miqcli/api.py

```diff
@@ -63,16 +68,79 @@ def connect(self, settings):
         :param settings: ManageIQ settings.
         :type settings: dict
         """
+        authfile = os.path.join(os.environ["HOME"], AUTHDIR)
+        # attempt a connection with a passed token
+        if "token" in settings and settings["token"]:
+            succ_auth, exception = self.miq_auth(settings["token"])
+            if not succ_auth:
+                raise RuntimeError("Unable to auth with passed token: "
+                                   "{0}".format(exception))
+        # check for an auth token file & validate
+        if os.path.isfile(authfile):
+            with file(authfile) as f:
```

Comment:
Could we use open instead of file? This provides Python 3.x compatibility.

```python
-            with file(authfile) as f:
+            with open(authfile) as f:
```